app/src/main/java/org/mozilla/fenix/compose/list/SectionHeader.kt

```diff
@@ -0,0 +1,99 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.list
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.PrimaryText
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Collapsible header for sections of lists
+ *
+ * @param sectionText The title of the section header.
+ * @param expanded Indicates whether the section of content is expanded. If null, the Icon will be hidden.
+ * @param onClick Optional lambda for handling section header clicks.
+ */
+@Composable
+fun SectionHeader(
+    sectionText: String,
+    expanded: Boolean? = null,
+    onClick: () -> Unit = {},
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable(onClick = onClick)
+            .padding(horizontal = 16.dp, vertical = 8.dp),
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        PrimaryText(
+            text = sectionText,
+            fontSize = 14.sp,
+            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
+            maxLines = 1,
+        )
+
+        expanded?.let {
+            Spacer(modifier = Modifier.width(8.dp))
+
+            Icon(
+                painter = painterResource(
+                    if (expanded) R.drawable.ic_chevron_down else R.drawable.ic_chevron_up
+                ),
+                contentDescription = stringResource(
+                    if (expanded) R.string.collapse_list_section else R.string.expand_list_section,
+                ),
+                modifier = Modifier.size(20.dp),
```

Comment:
This should be `24.dp`, but you might not even need this size modifier since the drawable is already `24dp`.

Reply:
Probably safer to add the `.size(24.dp)` when I think about it

Reply:
Can you reference what the node on figma? I am seeing 24x24 

https://www.figma.com/file/ujdiReypSypu2GhRgzArea/branch/ddGnMfYggXGmcTYfoI0aS8/Mobile-Android-Components?node-id=1839%3A57587
<img width="1680" alt="Screen Shot 2022-03-21 at 11 52 22 AM" src="https://user-images.githubusercontent.com/1190888/159299534-5c19457d-e75e-42f5-9fb9-2d768f6996e3.png">

We also don't have 20x20 mobile icons https://www.figma.com/file/Kv7wWr8vH6AQrGhu9vmjdE/Mobile-Icons?node-id=281%3A2094

Reply:
https://www.figma.com/file/ujdiReypSypu2GhRgzArea/branch/ddGnMfYggXGmcTYfoI0aS8/Mobile-Android-Components?node-id=1839%3A57587
<img width="647" alt="image" src="https://user-images.githubusercontent.com/87384386/159302222-1e610478-a7ea-4b1e-bf27-507e2e2b0761.png">

Reply:
Oh lol, I wasn't even looking at the right component